scripts/pgs/LDpred2/calculateLD.R

```diff
@@ -74,35 +84,38 @@ MAP <- obj.bigSNP$map
 GD <- snp_asGeneticPos(CHR, POS, dir=dirGeneticMaps, type=argGeneticMapsType, ncores=NCORES)
 if (is.null(GD)) stop('Genetic distance is not available')
 MAP <- MAP[,c('chromosome', 'marker.ID', 'physical.pos', 'allele1', 'allele2')]
-colnames(MAP) <- c('chr', 'rsid', 'pos', 'a1', 'a0')
 
-SNPs <- MAP$rsid
-if (!is.na(fileKeepSNPs)) {
-  cat('Reading SNPs from --file-keep-snps:', fileKeepSNPs, '\n')
-  keepSNPs <- read.table(fileKeepSNPs)[,1]
-  cat('Read', length(keepSNPs), 'SNPs\n')
-  SNPs <- SNPs[SNPs %in% keepSNPs]
-}
-if (!is.na(fileSumstats)) {
-  cat('Reading SNPs from sumstat file --sumstats:', fileSumstats, '\n')
-  dfSumStats <- data.table::fread(fileSumstats, sep="auto", data.table=F)
-  cat('Read', nrow(dfSumStats), 'SNPs\n')
-  SNPs <- SNPs[SNPs %in% dfSumStats[,columnRsidSumstats]]
-}
+# Filtering SNPs (if those arguments have been provided)
+SNPs <- MAP$marker.ID
+if (!is.na(fileSumstats)) SNPs <- filterFromFile(SNPs, fileSumstats, col=columnRsidSumstats)
+if (!is.na(fileKeepSNPs)) SNPs <- filterFromFile(SNPs, fileKeepSNPs)
+colnames(MAP) <- c('chr', 'rsid', 'pos', 'a1', 'a0')
 useSNPs <- MAP$rsid %in% SNPs
-cat('A total of', sum(useSNPs), 'will be used for LD calculation\n')
+nSNPs <- sum(useSNPs)
+if (nSNPs == 0) stop('No SNPs available.')
+cat('A total of', nSNPs, 'will be used for LD calculation\n')
 
 individualSample <- rows_along(G)
+# Extract individuals
+if (!is.na(extractIndividuals)) {
+  indIds <- filterFromFile(obj.bigSNP$fam, extractIndividuals, colFilter='sample.ID')
+  cat('Extracting ', sum(indIds), ' individuals\n')
```

Comment:
This line throws an error as `indIds` is the same object type as `obj.bigSNP$fam`. `sum` should be changed to `nrow` and on line 103 `indIds` should have been `indIds$fam$sample.ID`. Will change name of `indIds` to something better.
This was not detected in the tests and as far as I can see this parameter does not have a test. Will add that.